File: backend/go/backend-go-gin/internal/api/dto.go
package api

// Response bodies are structs rather than maps so the JSON field order matches the other
// reference backends exactly.

type healthResponse struct {
	Status  string `json:"status"`
	Service string `json:"service"`
}

type itemDTO struct {
	ID          int64  `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type itemsResponse struct {
	Items  []itemDTO `json:"items"`
	Source string    `json:"source"`
}

type authResponse struct {
	Token       string `json:"token"`
	Username    string `json:"username"`
	RedirectURL string `json:"redirectUrl"`
}

type profileResponse struct {
	Username string `json:"username"`
}

type errorResponse struct {
	Message string `json:"message"`
}

File: backend/rust/backend-rust-axum/src/api/dto.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct HealthResponse {
    pub status: &'static str,
    pub service: String,
}

#[derive(Serialize)]
pub struct ItemDto {
    pub id: i64,
    pub name: String,
    pub description: String,
}

#[derive(Serialize)]
pub struct ItemsResponse {
    pub items: Vec<ItemDto>,
    pub source: &'static str,
}

#[derive(Serialize)]
pub struct AuthResponse {
    pub token: String,
    pub username: String,
    #[serde(rename = "redirectUrl")]
    pub redirect_url: &'static str,
}

#[derive(Serialize)]
pub struct ProfileResponse {
    pub username: String,
}

#[derive(Serialize)]
pub struct ErrorResponse {
    pub message: String,
}
